File: test/examples/BadExample.cs
using System;
using System.Threading;

class BadExample {
    static string ApiKey = "abc123"; // exposed_secrets

    static int Factorial(int n) { // recursion
        return Factorial(n - 1);
    }

    static void Main() {
        var list = new List<int>(100); // dynamic_memory
        while (true) { // unbounded_loops
            if (list.Count > 0) {
                if (list[0] > 0) { // nested_conditionals
                    goto End; // complex_flow (goto)
                }
            }
            Console.WriteLine(Console.ReadLine()); // unsafe_input
            Thread.Sleep(1000); // set_timeout
            return; // complex_flow (multiple returns)
        End:
            yield return 1; // complex_flow (yield)
            return;
        }
    }
}

File: test/examples/bad-example.swift
import Foundation

let apiKey = "123abc" // exposed_secrets

func factorial(_ n: Int) -> Int { // recursion
    return factorial(n - 1)
}

var arr = Array(repeating: 0, count: 100) // dynamic_memory

func badExample() -> Int {
    while true { // unbounded_loops
        if arr[0] > 0 {
            if arr[1] > 0 { // nested_conditionals
                guard let input = readLine() else { fallthrough } // complex_flow (fallthrough), unsafe_input
            }
        }
        Thread.sleep(forTimeInterval: 1.0) // set_timeout
        return 1 // complex_flow (multiple returns)
    }
    do {
        throw NSError(domain: "", code: -1) // try_catch
        return 2
    } catch {
        print("Error") // insufficient_logging
    }
}